sdk/profiles/src/main/java/io/opentelemetry/sdk/profiles/data/AggregationTemporality.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.profiles.data;
+
+/**
+ * Specifies the method of aggregating metric values.
+ *
+ * @see "profiles.proto::AggregationTemporality"
+ */
+public enum AggregationTemporality {
+
+  /** The default AggregationTemporality, it MUST not be used. */
+  UNSPECIFIED,
```

Comment:
The profiling proto spec explicitly says it's fine to not set field aggregationTemporality in ValueType, so omitting an undefined value means allowing the aggregationTemporality field of ValueType to be null as an alternative way of indicating 'not set'? This is a case where copying comments from the .proto to javadoc is not so good - there is actually a potential use case for the enum value at Java level.

I'm also kind of fan of having the enum ordinals align with the protobuf ones, though to be fair not many users need to debug marshaling issues at that level.

Reply:
@jack-berg do you still want UNSPECIFIED gone, or shall we keep it to use in place of null / Optional<AggregationTemporality>, pending a decision on if the two AggregationTemporality definitions will be merged into one in common?

Reply:
I'd say we remove it in the interest of staying consistent with our existing AggregationTemporality concept. I think the most likely outcome is that the proto profiles AggregationTemporality definition ends up being the same as the metric definition. It will just be too difficult to rationalize have two definitions floating around the same thing. At that point we would want to follow suit. 

Where aggregation temporality is not actually optional for any metric data types which use it, it is for profiling, and those types can simply have `@Nullabe` return types to represent "not set".

Reply:
UNSPECIFIED gone, point of use now Nullable instead.